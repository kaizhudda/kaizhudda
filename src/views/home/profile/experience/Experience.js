import React from 'react';

import './Experience.scss';

const Experience = ({ children, title, companyName, period, location, companyLink, skills }) => {
  return (
    <div className="Experience col-xl-5 col-lg-6 col-md-9 col-xs-12">
      <div className="company">
        <h2>{title}</h2>
        <h3>{companyName}</h3>
        <p><i className="far fa-clock"></i>{period}</p>
      </div>
      <div className="desc">
        <p className="about">
          {children}
      </p>
      <p className="skills">
        <strong>Key Technologies: </strong> 
        {skills}
      </p>
        <div className="icons">
          <span className="location"><i class="fas fa-map-marker-alt" />{location}</span>
          <a href={`${companyLink}`}> <i className="fas fa-link"></i>{companyLink}</a>
        </div>
      </div>
    </div>
  );
}

export default Experience; 
