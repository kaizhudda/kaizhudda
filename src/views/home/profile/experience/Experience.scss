@import '../../../../styles//variables';
@import '../../../../styles/media-queries';

.Experience {
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: left;
  padding: 48px;
  background-color: rgba(white, 0.8);
  // box-shadow: 0 1.5rem 2rem rgba(#000000, 0.15);
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
  margin-bottom: 48px;
  width: 100%;
  border-radius: 3px;
  transition: all 0.3s;


  &:hover {
    transform: translateY(-0.5rem) scale(1.01);
  box-shadow: 0 1.5rem 2rem rgba(#000000, 0.15);
    // box-shadow: 0 2rem 2rem rgba(#000000, 0.2);
  }

  @include phone-only {
    padding: 16px;
  }

  .company {
    .far {
      margin-right: 5px;
    }

    h2 {
      font-family: $font-body;
      font-size: 1.5rem;
      margin-top: 0px;
      margin-bottom: 0px;
      font-weight: 700;
      color: #666;

      @include phone-only {
        font-size: 1.125rem;
      }
    }

    h3 {
      font-family: $font-body;
      font-size: 1.125rem;
      margin-top: 0px;
      margin-bottom: 8px;
      font-weight: 600;

      @include phone-only {
        font-size: 1rem;
      }
    }

    p {
      font-family: $font-body;
      font-size: 1rem;
      margin-top: 0px;
      font-weight: 500;
      color: #22a39f;

      @include phone-only {
        font-size: 0.925rem;
      }
    }
  }
  .desc {

    h3 {
      font-family: $font-body;
      font-size: 1.125rem;
      margin-top: 0px;
      margin-bottom: 5px;
      font-weight: 700;
    }

    p {
      line-height: 1.60;
      font-family: $font-body;
      font-size: 1rem;

      @include phone-only{
        font-size: 0.925rem;
      }
    }

    .skills {
      display: flex;
      flex-direction: column;
      // justify-content: center;
      // align-items: center;
    }

    .key-tech-title {
      font-family: $font-body;
      font-size: 1.05rem;
      margin-top: 40px;
      margin-bottom: 0px;
      font-weight: 600;
    }

    .tech-list {
      display: flex;
      flex-wrap: wrap;
      align-items: center;

      .icon-container {
        display: flex;
        flex-direction: column;
        padding: 16px 16px 4px 0px;
        align-items: center;

        .tech-title {
          text-align: center;
          font-size: 0.875rem;
        }

        .redux-icon {
          height: 32px;
          margin-bottom: 8px;
        }

        .webpack-icon {
          height: 38px;
          margin-bottom: 2px;
        }
      }

      .fab {
        font-size: 2.25rem;
        margin-bottom: 4px;
      }

      .fa-react {
        color: #00d8ff;
      }

      .fa-js {
        color: #f0db4f;
      }

      .fa-html5 {
        color: #e34c26;
      }

      .fa-css3-alt { 
        color: #31a1d9;
      }

      .fa-git-alt {
        color: #f34f29;
      }
    }
  }

  .icons {
    display: flex;
    @include phone-only {
      flex-direction: column;
    }
    .fas {
      color: #22a39f;
      margin-right: 5px; 
    }
    .location {
      margin-right: 10px;
      font-family: $font-body;
      font-size: 1rem;
      color: #22a39f;

      @include phone-only {
        margin-bottom: 5px;
        font-size: 0.925rem;
      }
    }
    a {
      color: #22a39f;
      text-decoration: none;
      font-family: $font-body;
      font-size: 1rem;

      &:hover {
        text-decoration: underline;
      }

      @include phone-only {
        font-size: 0.925rem;
      }
    }
  }
}