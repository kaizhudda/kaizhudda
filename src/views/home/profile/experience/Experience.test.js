import React from 'react';
import { shallow } from 'enzyme';
import Experience from './Experience';

let props = {
  title: 'Test Title',
  companyName: 'Test Company',
  period: 'Feb - Present',
  location: 'Auckland, NZ',
  companyLink: 'Link.com',
  skills: 'Jest, Enzyme'
};
const children = 'Test Bullet points';
let wrapped = shallow(<Experience  {...props}>{children}</Experience>); 

describe('Experience', () => {
  it('should render the Experience Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render the title', () => {
    expect(wrapped.find('h2').text()).toEqual(props.title);
  });

  it('should render the company Name', () => {
    expect(wrapped.find('h3').text()).toEqual(props.companyName);
  });

  it('should render the period', () => {
    expect(wrapped.find('p').first().text()).toEqual(props.period);
  });

  it('should render the children correctly', () => {
    expect(wrapped.find('.about').text()).toEqual(children);
  });

  it('should render the location correctly', () => {
    expect(wrapped.find('.location').text()).toEqual(props.location);
  });

  it('should render the company Link correctly', () => {
    expect(wrapped.find('a').getElement().props.href).toEqual(props.companyLink);
    expect(wrapped.find('a').text()).toEqual(props.companyLink);
  });

  it('should render the skills correctly', () => {
    expect(wrapped.find('.skills').text()).toEqual(`Key Technologies: ${props.skills}`);
  });

  describe('when education prop is true', () => {
    beforeEach(() => {
      props = {
        ...props,
        education: true
      }
      wrapped = shallow(<Experience  {...props}>{children}</Experience>); 

    });
    it('should not render skills when eduction flag is true', () => {
      expect(wrapped.find('.skills').exists()).toBe(false);
    });
  });
});
