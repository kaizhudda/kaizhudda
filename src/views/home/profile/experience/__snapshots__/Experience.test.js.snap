// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience should render the Experience Component correctly 1`] = `
<div
  className="Experience col-xl-5 col-lg-6 col-md-9 col-xs-12"
>
  <div
    className="company"
  >
    <h2>
      Test Title
    </h2>
    <h3>
      Test Company
    </h3>
    <p>
      <i
        className="far fa-clock"
      />
      Feb - Present
    </p>
  </div>
  <div
    className="desc"
  >
    <p
      className="about"
    >
      Test Bullet points
    </p>
    <p
      className="skills"
    >
      <strong>
        Key Technologies: 
      </strong>
      Jest, Enzyme
    </p>
    <div
      className="icons"
    >
      <span
        className="location"
      >
        <i
          className="fas fa-map-marker-alt"
        />
        Auckland, NZ
      </span>
      <a
        href="Link.com"
      >
        <i
          className="fas fa-link"
        />
        Link.com
      </a>
    </div>
  </div>
</div>
`;
